frontends/p4/def_use.h

```diff
@@ -104,22 +121,27 @@ class BaseLocation : public StorageLocation {
 
 /// Base class for location sets that contain fields
 class WithFieldsLocation : public StorageLocation {
- protected:
-    hvec_map<cstring, const StorageLocation *> fieldLocations;
-    friend class StorageFactory;
-    WithFieldsLocation(const IR::Type *type, cstring name) : StorageLocation(type, name) {}
+    flat_map<cstring, const StorageLocation *, std::less<>,
+             absl::InlinedVector<std::pair<cstring, const StorageLocation *>, 4>>
```

Comment:
I'm very surprised that a flat_map is faster than an hvec_map here -- for that to be true, almost all the structs/headers in the program have no more than 2-3 fields?

Reply:
It is not about "fast". It is about memory consumption. Here we do not allocate memory at all for any structs with 4 or less fields. And we resort to memory allocation for anything more. But yes, normally small flat maps have comparable speed to hash maps. But yes, as far as I can see, most structs are small. There are certainly large ones but they are much rare. And we normally do not have struct larger than, say, 16 fields.